src/Utilities/FlowAnalysis/FlowAnalysis/Framework/DataFlow/DataFlowOperationVisitor.cs

```diff
@@ -2714,7 +2714,10 @@ out HashSet<(IMethodSymbol method, IOperation instance)> resolvedMethodTargetsOp
             else
             {
                 resolvedMethodTargetsOpt = null;
-                ResetCurrentAnalysisData();
+                if (PessimisticAnalysis)
```

Comment:
Ah nice catch. Do we have a unit test to stop me from breaking this again :) ?

Reply:
I forget which one in particular, but I found this from one of the new unit tests I wrote.